src/core/reference/src/op/random_uniform.cpp

```diff
@@ -158,8 +126,9 @@ void convert_to_output_type(const std::vector<uint32_t>& res,
     std::vector<T> res_out_type(step);
     if (elem_type.size() > 4) {
         // Each element of resulting sequence is formed using two uint32 values
-        res_out_type[0] = convert_two_inputs(res[0], res[1], mn[0], mx[0]);
-        res_out_type[1] = convert_two_inputs(res[2], res[3], mn[0], mx[0]);
+        for (size_t i = 0; i < step / 2; ++i) {
```

Comment:
Size is guaranteed to be one of the {2, 4, 8} as the only acceptable types of RandomUniform are int32, int64, bflat16, float16, float32, float64.

Yes, half of the values set is okay, since we consume 2 uint32 values to create a 64-bit value (guaranteed since elem_type.size() > 4) we generate half of the amount of step numbers. (step - amount of generated 32-bit numbers by generator)

This function was already in the original code, so I presume that it works as intended, and the tests reflect that. The code style is a bit messy, I can rename some variables for readability